### ISSUE TITLE
Cleanup compaction filters out sstables that are being compacted

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#9257]
Similar to #9256, cleanup compaction calls `get_candidates` without stopping ongoing compactions.
https://github.com/scylladb/scylla/blob/7f44736939eca788100287e7ff2150ce55e64483/compaction/compaction_manager.cc#L839

Similar to scrub, regular compaction will not cleanup tokens that do not longer belong to the current node and we don't want to leave those around after cleanup.  From the use standpoint once cleanup is done successfully after nodes are added or removed, all tokens are cleaned up and there's no risk of data resurrection, in case ownership of some tokens moves back to the original node.

I'm opening a separate issue so this can be fixed and backported independently of #9256.

**Comment by raphaelsc** [comment]
This is a dup of https://github.com/scylladb/scylla/issues/8168

closing
